### PR TITLE
[Angular] Loop for enums

### LINKED ISSUE #16407: Angular: Loop for enums
##### **Overview of the feature request**

Currently the HTML part of an entity is generated listing all the possible values of an enum:

```html
<div class="form-group">
    <label class="form-control-label" jhiTranslate="myApp.myEntity.drink" for="field_drink">Drink</label>
    <select class="form-control" name="drink" formControlName="drink" id="field_drink" data-cy="drink">
        <option [ngValue]="null">{{ 'myApp.Drink.null' | translate }}</option>
        <option value="WATER">{{ 'myApp.Drink.WATER' | translate }}</option>
        <option value="TEE">{{ 'myApp.Drink.TEA' | translate }}</option>
        <option value="COFFEE">{{ 'myApp.Drink.COFFEE' | translate }}</option>
    </select>
</div>
```

It would be nice if this could be changed using a loop:

```html
<div class="form-group">
    <label class="form-control-label" jhiTranslate="myApp.myEntity.drink" for="field_drink">Drink</label>
    <select class="form-control" name="drink" formControlName="drink" id="field_drink" data-cy="drink">
        <option [ngValue]="null">{{ 'myApp.Drink.null' | translate }}</option>
        <option *ngFor="let drink of drinks" [ngValue]="drink">{{ 'myApp.Drink.' + drink | translate }}</option>
    </select>
</div>
```

This requires this code to be present in the TypeScript part:

```typescript
import { Drink } from 'app/entities/enumerations/drink.model';
// ...

export class SomeComponent implements OnInit {
  drinks = Object.keys(Drink);
  // ...
}
```

Maybe this can also be applied to React.

##### **Motivation for or Use Case**

Better maintainability of the enum code.

##### **Related issues or PR**

- [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by qmonmert**
Fix #16407 

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Review by DanielFran — APPROVED** [review]

**Review by mshima — APPROVED** [review]
A suggestion is to use `<%- _.lowerFirst(importedType) %>Values` instead of `<%- importedType %>s`, since it's not a class and would be more descriptive.
But looks good as it is, your call.

**Comment by qmonmert** [comment]
@mshima done

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/update/entity-management-update.component.html.ejs [hunk 1]
```diff
@@ -54,10 +54,8 @@ _%>
         <%_ const enumPrefix = frontendAppName + '.'+ fieldType; _%>
                         <option [ngValue]="null"><% if (enableTranslation) { %>{{ '<%= enumPrefix %>.null' | translate }}<% } %></option>
         <%_ const values = field.enumValues;
-            for (key in values) {
-                const enumValue = values[key]; _%>
-                        <option value="<%= enumValue.name %>"><% if (enableTranslation) { %>{{ '<%= enumPrefix %>.<%= enumValue.name %>' | translate }}<% } else { %><%= enumValue.value %><% } %></option>
-        <%_ } _%>
+            const enumImports = generateEntityClientEnumImports(fields); _%> 
+                        <option *ngFor="let <%= fieldType %> of <%- _.lowerFirst(fieldType) %>Values" [value]="<%= fieldType %>"><% if (enableTranslation) { %>{{ '<%= enumPrefix %>.' + <%= fieldType %> | translate }}<% } else { %><%= fieldType %><% } %></option>
```
Comment (mshima) [inline generators/entity-client/templates/angular/src/main/webapp/app/entities/update/entity-management-update.component.html.ejs:58]:
```suggestion
                        <option *ngFor="let <%= _.lowerFirst(fieldType) %> of <%- _.lowerFirst(fieldType) %>Values" [value]="<%= _.lowerFirst(fieldType) %>"><% if (enableTranslation) { %>{{ '<%= enumPrefix %>.' + <%= fieldType %> | translate }}<% } else { %><%= fieldType %><% } %></option>
```

**Review by mshima — COMMENTED** [review]
Awesome.

**Comment by qmonmert** [comment]
@mshima done :)

**Pull request merged**